src/backend/base/langflow/components/processing/parse.py

```diff
@@ -0,0 +1,167 @@
+import json
+from typing import Any
+
+from langflow.custom import Component
+from langflow.helpers.data import data_to_text, data_to_text_list
+from langflow.io import (
+    DataFrameInput,
+    DataInput,
+    DropdownInput,
+    MultilineInput,
+    Output,
+    StrInput,
+)
+from langflow.schema import Data, DataFrame
+from langflow.schema.message import Message
+
+DATAFRAME_ERROR = "Expected a DataFrame"
+DATA_ERROR = "Expected Data object(s)"
+
+
+class ParseComponent(Component):
+    """Parse DataFrames or Data objects into text using templates or default formatting."""
+
+    display_name = "Parse"
+    description = "Parse DataFrames or Data objects into text using templates or default formatting."
+    icon = "braces"
+    name = "Parse"
+
+    inputs = [
+        DropdownInput(
+            name="parse_input_type",
```

Comment:
Would it be possible instead to have a `HandleInput` that accepts both `Data` and `DataFrame` as `input_types` ?

Reply:
I think it's easier to have an input called "Data/DataFrame" that you just connect instead of a dropdown where you need to click to discover what's in there and will change the aspect of your component dynamically.